recipes/default.rb

```diff
@@ -217,13 +216,37 @@
   mode 0750
   variables(
     {
-      :kafka_fqdn => kafka_fqdn,
       :mgm_fqdn => mgm_fqdn,
       :hopsworks_url => hopsworks_url
     }
   )
 end
 
+ruby_block 'copy-config-dir' do
+  block do
+    require 'fileutils'
+
+    if !node['onlinefs']['config_dir'].nil?
+      # Copy everything from the provided config_dir to etc overwriting any duplicates
+      FileUtils.cp_r(Dir["#{node['onlinefs']['config_dir']}/*"], node['onlinefs']['etc'])
+    end
+  end
+end
+
+kafka_fqdn = consul_helper.get_service_fqdn("broker.kafka")
+template "#{node['onlinefs']['etc']}/#{['onlinefs']['kafka']['properties_file']}" do
+  source "onlinefs-kafka.properties.erb"
+  owner node['onlinefs']['user']
+  group node['onlinefs']['group']
+  mode 0750
+  action :create_if_missing
```

Comment:
We can't use `:create_if_missing` because we will never be able to update the values of the configuration. 

You can either add an `not_if` comment or swap the 2